api/src/services/organizations/organizations.ts

```diff
@@ -24,6 +25,44 @@ export const createOrganization: MutationResolvers['createOrganization'] = ({
   })
 }
 
+export const getOrCreateOrganization = async (orgName, reportingPeriodName) => {
+  try {
+    let orgRecord
+
+    const existingOrganization = await db.organization.findFirst({
+      where: { name: orgName },
+    })
+    if (existingOrganization) {
+      logger.info(`Organization ${orgName} already exists`)
+      orgRecord = existingOrganization
+    } else {
+      logger.info(`Creating ${orgName}`)
+      const data: Prisma.OrganizationCreateArgs['data'] = {
+        name: orgName,
+      }
+      const reportingPeriod = await db.reportingPeriod.findFirst({
+        where: { name: reportingPeriodName },
+      })
+      if (!reportingPeriod) {
+        logger.error(
+          `Reporting period ${reportingPeriodName} does not exist. Cannot create organization.`
+        )
+        return
+      }
+      data.preferences = {
+        current_reporting_period_id: reportingPeriod.id,
+      }
+      orgRecord = await db.organization.create({
+        data,
+      })
+    }
+    return orgRecord
+  } catch (error) {
+    logger.error(`Error creating organization: ${orgName}`)
+    logger.error(error)
+  }
```

Comment:
Using this as an example, although I think it applies to the other get-or-create service functions:

The try/catch handling here seems overly broad, as the `try` block includes  the retrieval logic as well as the creation logic, meaning that if an error is thrown during retrieval (and therefore the create operation never runs), the error message will be `Error creating organization` which is misleading. 

These functions only seem intended for use by the seed script so maybe it doesn't matter quite as much, but the error-handling in general feels like error suppression in that there's no path to recovery afforded to any callers. I think it would be better to allow any error thrown in the course of executing this function to bubble up to the caller, with logging and/or recovery (if possible) happening there. Right now, the log message with no accompanying `throw` means that the only way we'd know about a problem is if we happened to spot it in the logs, which stops this function from being applicable outside of cases where a human is actively monitoring execution (which, if they're not intended to be reusable, makes me wonder if these get-or-create functions should just live in the seeds file).

Reply:
> which, if they're not intended to be reusable, makes me wonder if these get-or-create functions should just live in the seeds file
These functions are only intended to be executed through human intervention. The reason for it to reside in these `services` files is for ease of unit-testing since these files have the right testing infrastructure already setup. The aim in the next few quarters is to not need the script since all these actions _should_ be able to be executed via the front-end through CRUD operations.

> The try/catch handling here seems overly broad,
I intended each of these functions to be self-contained and not block the script execution. Since there are many operations that are getting executed simultaneously and we don't want to halt progress if only one of the entities had an issue.

> meaning that if an error is thrown during retrieval (and therefore the create operation never runs), the error message will be Error creating organization which is misleading.
Agreed, will update the error message to be clarify that it could be during retrieval as well.